src/main/java/org/opensearch/security/auth/UserInjector.java

```diff
@@ -57,14 +59,13 @@
     private final ThreadPool threadPool;
     private final AuditLog auditLog;
     private final XFFResolver xffResolver;
-    private final Boolean injectUserEnabled;
+    private final Boolean injectRESTUserEnabled;
```

Comment:
I don't see a need to restrict this just to REST API. User injection is either globally enabled or globally disabled on the cluster.

Reply:
We don't want to explicitly block on the user injection support (as with role injection) for the transport requests. Note that, this setting was enabled previously for only the REST requests and we added support for transport request afterwards.

With other plugins depending on user injection and using this settings for the transport requests , this should be enabled by default.

Reply:
It may be enabled by default, but cluster admin must have an option not to trust user injection from other plugins/nodes. I don't see a need to have a separate settings for REST and transport that would allow to disable one type of injection and enable another.

Reply:
I will move the setting to be enabled by default and move the everything under this setting. Let me know, If this addresses your concern.

Reply:
The default should be the most secure option. On a preconfigured cluster that has multiple plugins installed, the settings can be explicitly enabled.